hacspec/src/array.rs

```diff
@@ -408,67 +408,67 @@ macro_rules! generic_array {
             }
         }
 
-        impl<T: Numeric> Index<usize> for $name<T> {
+        impl<T: Numeric + Copy> Index<usize> for $name<T> {
```

Comment:
Shouldn't `Numeric` implement `Copy` in all cases ?

Reply:
Well, so this is probably the most controversial change here.
The problem is that `Integer` is a `Numeric` and we need that property only in some areas. And if we want to have matrices etc. we would need `Numeric` there. Also `BigInt` are obviously not `Copy`... So I'm not really happy with that change either. Happy for suggestions :thinking:

Reply:
Ok. Indeed `Numeric` and `Copy` should be orthogonal. For instance we would want `ByteSeq` (which is not `Copy`) to implement `Numeric`. So let's keep both distinct !